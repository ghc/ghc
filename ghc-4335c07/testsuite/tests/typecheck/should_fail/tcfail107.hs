-- The type sig for 'test' is illegal in H98 because of the
-- partial application of the type sig.
-- But with the LiberalTypeSynonyms extension enabled it
-- should be OK because when you expand the type synonyms
-- it's just Int->Int
-- c.f should_compile/tc155.hs
module ShouldFail where

type Thing m = m ()

type Const a b = a

test :: Thing (Const Int) -> Thing (Const Int)
test = test
