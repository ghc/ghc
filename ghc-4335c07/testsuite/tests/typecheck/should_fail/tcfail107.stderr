
tcfail107.hs:13:9: error:
    • The type synonym ‘Const’ should have 2 arguments, but has been given 1
    • In the type signature:
        test :: Thing (Const Int) -> Thing (Const Int)
